src/app/clusters/commodity-price-server/commodity-price-server.cpp

```diff
@@ -511,6 +487,136 @@ CHIP_ERROR Instance::SetForecast(const DataModel::List<const Structs::CommodityP
     return CHIP_NO_ERROR;
 }
 
+void Instance::CheckAndFreeForecastBuffers()
+{
+    if (mPriceForecast.size() == 0)
+    {
+        VerifyOrDie(mOwnedForecastPriceStructBuffer.Get() == nullptr);
```

Comment:
Similar issue here: if a previous components copy failed, this will now crash...  Followup is OK to make this safer, but we should do it.